newIDE/app/src/MainFrame/index.js

```diff
@@ -3086,6 +3091,9 @@ const MainFrame = (props: Props) => {
           onClose={languageChanged => {
             openPreferencesDialog(false);
             if (languageChanged) _languageDidChange();
+            if (app && preferences.values.newProjectsDefaultFolder === '') {
+              preferences.setNewProjectsDefaultFolder(findDefaultFolder(app));
```

Comment:
You're putting this logic inside MainFrame, but it should not be MainFrame responsibility to do it. Think of it:
MainFrame is a place where all the editors are tied together. If we start adding code that is tied to a specific feature, MainFrame will soon be full of code that does everything.

I also think this does not solve the issue that is being talked :)
Come back to the drawing board ;)